### PR TITLE
Separate memory index for Pinecone, Redis and local cache memory backends

### PR DESCRIPTION
**Pull request by Tymec**
### Background

This change proposes to replace `memory_index` which is used in both Redis and LocalCache memory backends, with separate config options for each memory backend. This would make the code more flexible and configurable, as well as consistent with other memory backends.

### Changes

- Create `redis_index`, `local_cache_index` and `pinecone_index` config options for each memory backend
- Remove `memory_index` config option
- Remove hardcoded table name in Pinecone and replace it with `pinecone_index`

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]
Isn't it better to use separate indexes for Redis and Pine? What if we design an architecture where 2 types of memory are possible at the same time with different index names?

**Comment by Tymec** [comment]
Since both Redis and Local memory types shared the same memory index, I assumed that's what was meant. Having each memory type have it's own memory index does sound like a better overall approach. Either way, I implemented the changes and now each memory type (except NoMemory) should have it's own index.

**Comment by nponeccop** [comment]
@Tymec There are conflicts now

**Review by nponeccop — DISMISSED** [review]

**Review by richbeales — DISMISSED** [review]

**Comment by richbeales** [comment]
Run flake8 autogpt/ tests/ --select E303,W293,W291,W292,E305,E231,E302
autogpt/config/config.py:85:1: W293 blank line contains whitespace

**Review by nponeccop — DISMISSED** [review]

**Review by richbeales — DISMISSED** [review]

**Comment by richbeales** [comment]
Conflicting files
autogpt/config/config.py
autogpt/memory/redismem.py

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

File: .env.template
Comment (lc0rp) [inline .env.template:1]: Did you want to add LOCAL_CACHE_INDEX to the env file?

File: autogpt/memory/pinecone.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __init__(self, cfg):
         dimension = 1536
         metric = "cosine"
         pod_type = "p1"
-        table_name = "auto-gpt"
+        table_name = cfg.pinecone_index
```
Comment (lc0rp) [inline autogpt/memory/pinecone.py:17]: 👍

**Pull request closed without merging**